x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/cat/RestCatJobStatsAction.java

```diff
@@ -0,0 +1,389 @@
+
+    private Table buildTable(RestRequest request, GetJobsStatsAction.Response jobStats) {
+        Table table = getTableWithHeader(request);
+        jobStats.getResponse().results().forEach(job -> {
+            table.startRow();
+            table.addCell(job.getJobId());
+            table.addCell(job.getState().value());
+            table.addCell(job.getOpenTime());
+            table.addCell(job.getAssignmentExplanation());
+
+            DataCounts dataCounts = job.getDataCounts();
+            table.addCell(dataCounts.getProcessedRecordCount());
+            table.addCell(dataCounts.getProcessedFieldCount());
+            table.addCell(new ByteSizeValue(dataCounts.getInputBytes()));
+            table.addCell(dataCounts.getInputRecordCount());
+            table.addCell(dataCounts.getInputFieldCount());
+            table.addCell(dataCounts.getInvalidDateCount());
+            table.addCell(dataCounts.getMissingFieldCount());
+            table.addCell(dataCounts.getOutOfOrderTimeStampCount());
+            table.addCell(dataCounts.getEmptyBucketCount());
+            table.addCell(dataCounts.getSparseBucketCount());
+            table.addCell(dataCounts.getBucketCount());
+            table.addCell(dataCounts.getEarliestRecordTimeStamp());
+            table.addCell(dataCounts.getLatestRecordTimeStamp());
+            table.addCell(dataCounts.getLastDataTimeStamp());
+            table.addCell(dataCounts.getLatestEmptyBucketTimeStamp());
+            table.addCell(dataCounts.getLatestSparseBucketTimeStamp());
+
+
+            ModelSizeStats modelSizeStats = job.getModelSizeStats();
+            table.addCell(modelSizeStats == null ? null : new ByteSizeValue(modelSizeStats.getModelBytes()));
+            table.addCell(modelSizeStats == null ? null : modelSizeStats.getMemoryStatus().toString());
+            table.addCell(modelSizeStats == null || modelSizeStats.getModelBytesExceeded() == null ?
+                null :
+                new ByteSizeValue(modelSizeStats.getModelBytesExceeded()));
+            table.addCell(modelSizeStats == null || modelSizeStats.getModelBytesMemoryLimit() == null ?
+                null :
+                new ByteSizeValue(modelSizeStats.getModelBytesMemoryLimit()));
+            table.addCell(modelSizeStats == null ? null : modelSizeStats.getTotalByFieldCount());
+            table.addCell(modelSizeStats == null ? null : modelSizeStats.getTotalOverFieldCount());
+            table.addCell(modelSizeStats == null ? null : modelSizeStats.getTotalPartitionFieldCount());
+            table.addCell(modelSizeStats == null ? null : modelSizeStats.getBucketAllocationFailuresCount());
+            table.addCell(modelSizeStats == null ? null : modelSizeStats.getLogTime());
+            table.addCell(modelSizeStats == null ? null : modelSizeStats.getTimestamp());
+
+            ForecastStats forecastStats = job.getForecastStats();
+            table.addCell(forecastStats == null ? null : forecastStats.getTotal());
```

Comment:
nit: a bool to avoid repeating `forecastStats == null || forecastStats.getTotal() <= 0L` for every cell?

Reply:
For sure :). I was in fast editing mode (`yy 7p`) and tons of line duplications.